tests/drivers/disk/src/main.c

```diff
@@ -0,0 +1,219 @@
+/*
+ * Copyright (c) 2021 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/*
+ * WARNING: This test will overwrite data on any disk utilized. Do not run
+ * this test with an disk that has useful data
+ */
+
+#include <zephyr.h>
+#include <ztest.h>
+#include <storage/disk_access.h>
+#include <device.h>
+
+#if defined(CONFIG_SDMMC_USDHC)
```

Comment:
Please change it to:
```diff
diff --git a/tests/drivers/disk/src/main.c b/tests/drivers/disk/src/main.c
index b276b649085..f9bcbb7b99e 100644
--- a/tests/drivers/disk/src/main.c
+++ b/tests/drivers/disk/src/main.c
@@ -14,8 +14,10 @@
 #include <storage/disk_access.h>
 #include <device.h>
 
-#if defined(CONFIG_SDMMC_USDHC)
-#define DISK_NAME "SDMMC"
+#if IS_ENABLED(CONFIG_DISK_DRIVER_SDMMC)
+#define DISK_NAME CONFIG_SDMMC_VOLUME_NAME
+#elif IS_ENABLED(CONFIG_DISK_DRIVER_RAM)
+#define DISK_NAME CONFIG_DISK_RAM_VOLUME_NAME
 #else
 #error "No disk device defined, is your board supported?"
 #endif

```

regarding CONFIG_DISK_DRIVER_RAM, I tested your PR on top of https://github.com/zephyrproject-rtos/zephyr/pull/40611